nginx-controller/nginx/configurator.go

```diff
@@ -766,6 +786,45 @@ func (cnf *Configurator) updatePlusEndpoints(ingEx *IngressEx) error {
 	return nil
 }
 
+func filterMasterAnnotations(annotations map[string]string) (map[string]string, []string) {
+	var removedAnnotations []string
+
+	for _, blacklistAnn := range masterBlacklist {
+		if _, ok := annotations[blacklistAnn]; ok {
+			removedAnnotations = append(removedAnnotations, blacklistAnn)
+			delete(annotations, blacklistAnn)
+		}
+	}
+
+	return annotations, removedAnnotations
+}
+
+func filterMinionAnnotations(annotations map[string]string) (map[string]string, []string) {
+	var removedAnnotations []string
+
+	for _, blacklistAnn := range minionBlacklist {
+		if _, ok := annotations[blacklistAnn]; ok {
+			removedAnnotations = append(removedAnnotations, blacklistAnn)
+			delete(annotations, blacklistAnn)
+		}
+	}
+
+	return annotations, removedAnnotations
+}
+
+func mergeMasterAnnotationsIntoMinion(minionAnnotations map[string]string, masterAnnotations map[string]string) (map[string]string, []string) {
+	var minionOverwrites []string
+
+	for key, val := range masterAnnotations {
+		if _, ok := minionAnnotations[key]; !ok {
+			minionAnnotations[key] = val
+			minionOverwrites = append(minionOverwrites, key)
+		}
+	}
+
+	return minionAnnotations, minionOverwrites
```

Comment:
I don't think we need to return minionOverwrites.
Moreover, this function should not return anything, because the map minionAnnotations is passed by reference and will be changed by the function.